### DISCUSSION TITLE
what's the relationship between aws application composer and sam cli?

### DISCUSSION THREAD

**Discussion by lgtminshi** [aws/aws-sam-cli#4507] [Questions] [answered]
Through the internet infering, aws cloud acquired stackery.io at 2021, then the developers from stackery.io developed sam pipelines,
recently aws cloud published application composer(predecessor comes from stackery.io). User migrates from serverful to serverless or serverless in-use, writing the IaC file is not a big obstacle. It looks like  application composer’s role is relatively limited. 
Though It's completely bi-directional and it interfaces nicely with your local system. Whatever, the previous experience is enough for me using vscode sam extension.

**Comment by ryanycoleman** [comment, marked as answer]
Hello @lgtminshi and thank you for your question! 

I lead product for both of these projects. We think of Application Composer and the SAM CLI as complementary tools for building serverless applications. Composer’s connected mode enables it to work seamlessly, through the browser,  with any local tool you have installed that understands CloudFormation/SAM projects. For example, I often use Composer with `sam sync --watch` running in my IDE so that resources I drag onto the canvas are provisioned in my account by the time I need to work with them in my IDE. Here’s a gif demonstrating Composer/IDE connectivity if you haven’t tried it yet. 

![8  use-your-ide-and-tool](https://user-images.githubusercontent.com/1148878/210876332-c50f7bf2-fdbb-41b4-b312-4a7558d23a44.gif)

What would you like to see from a combination of Composer, SAM CLI, and your IDE?
